src/core/Akka/Dispatch/ConcurrentQueueMailbox.cs

```diff
@@ -69,7 +69,7 @@ private void Run()
                     Mailbox.DebugPrint(ActorCell.Self + " processing message " + envelope);
 
                     //run the receive handler
-                    ActorCell.Invoke(envelope);
+                    ActorCell.Dispatcher.Dispatch(ActorCell, envelope);
```

Comment:
This is for aligning more to JVM Akka.
The Dispatcher can execute extra logic before processing each message.

The Mailbox should probably not know about the owner cell but the dispatcher when port is complete.